### PR TITLE
Update commit from trevoz/lnet

### PR DESCRIPTION
**Pull request by zamronypj**
This PR contains work from [trevoz/lnet](https://github.com/trevoz/lnet) also minor fix regarding FPC version IFDEF.

Relevant discussion can be read [here](https://forum.lazarus.freepascal.org/index.php/topic,58762.0.html).

### CODE REVIEW

**Comment by almindor** [comment]
Thank you, this looks great on first glance. I'll have to test things out before merging tho. Hopefully in a few days or so.

**Comment by almindor** [comment]
Thank you, this looks great on first glance. I'll have to test things out before merging tho. Hopefully in a few days or so.

File: lib/lcommon.pp [hunk 1]
```diff
@@ -461,23 +461,28 @@ function IsPipeError(const anError: Integer): Boolean; inline;
 
 function TZSeconds: Integer; inline;
 begin
-  Result := unixutil.TZSeconds;
+{$IF FPC_FULLVERSION >= 33100}
+  Result := TZInfo.Seconds;
+{$ENDIF}
```
Comment (almindor) [inline lib/lcommon.pp]: could you make this into an `else`? otherwise we get a warning here since the compiler cannot infer that it's a full set

**Review by almindor — CHANGES_REQUESTED** [review]
just a minor nit, otherwise this looks great!

File: lib/lcommon.pp (hunk 1 above)
Comment (zamronypj) [reply lib/lcommon.pp]: @almindor  Ok I will look into it

Comment (zamronypj) [reply lib/lcommon.pp]: @almindor  It looks like @trevoz uses wrong `FPC_FULLVERSION`. For Free Pascal 3.3.1 or newer value should be `{$IF FPC_FULLVERSION >= 30301}` while for Free Pascal  3.3.0 or older `{$IF FPC_FULLVERSION <= 30300}`.

[Read FPC_FULLVERSION documentation here](https://www.freepascal.org/docs-html/prog/progap7.html).

**Review by almindor — APPROVED** [review]

**Comment by almindor** [comment]
Thank you!

**Pull request merged**